azurelinuxagent/common/protocol/metadata_server_migration_util.py

```diff
@@ -77,3 +78,102 @@ def _ensure_file_removed(directory, file_name):
     path = os.path.join(directory, file_name)
     if os.path.isfile(path):
         os.remove(path)
+
```

Comment:
The way the code for cleaning up the metadata is written, it needs to run on the daemon/pa. 

I moved the firewall code from common/ostutil to ga/FirewallManager. There should not be dependencies across the pa/ga code, so I am duplicating a little bit of code here. That code comes practically unchanged from osutil.

This clean up is/was needed when migrating Azure Stack VMs to Azure. Azure Stack used to have its own version of the WireServer (it was named the Metadata protocol) and some differences between those protocols require this cleanup. I do not know if this code is still needed, but it is owned and tested by the Azure Stack team.